torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -353,6 +353,47 @@ Tensor _nested_from_padded_backward(
   return grad.to_padded_tensor(0, input.sizes());
 }
 
+std::tuple<Tensor, Tensor, Tensor> NestedTensor_linear_backward(
+    const Tensor& grad,
+    const Tensor& self,
+    const Tensor& weight,
+    const c10::optional<Tensor>& bias_opt) {
+  if (!grad.defined()) {
+    return std::tuple<Tensor, Tensor, Tensor>{Tensor(), Tensor(), Tensor()};
+  }
+  auto* nt_grad = at::native::get_nested_tensor_impl_or_null(grad);
+  auto* nt_self = at::native::get_nested_tensor_impl_or_null(self);
```

Comment:
Just to be safe, we could add a torch internal assert that these ptr is not nullptr

Reply:
Another way of achieving that is by using the `is_nested` method. This isn't unlike the device and dtype checks we're doing everywhere.